plastron/auth/auth.py

```diff
@@ -0,0 +1,179 @@
+import time
+
+from abc import ABC, abstractmethod
+from jwcrypto.jwk import JWK  # type: ignore
+from jwcrypto.jwt import JWT  # type: ignore
+import logging
+from requests import Session
+from typing import Dict, List, Type
+
+
+class Auth(ABC):
+    @abstractmethod
+    def __init__(self, config: Dict[str, str]):
+        pass
```

Comment:
I added this because without it, "mypy" would complain:

```
>  mypy --strict plastron/auth/auth.py
plastron/auth/auth.py:58: error: Too many arguments for "Auth"
```

I think I can satisfy the IDE by calling "super.__init__(config)" from each of the subclasses.

Reply:
Yeah, that would also work. I am not as used to the strict typing enforcement in Python. It is definitely helpful in some cases, but I also feel like it can get in the way of some of the readability of Python code, just for the sake of satisfying the tool.

But in this case, I think you are correct that the proper OO way is to have the subclasses call their `super().__init__(config)`.